src/Microsoft.OData.Core/JsonLight/ODataJsonLightDeserializer.cs

```diff
@@ -908,6 +1099,369 @@ private void ReadODataOrCustomInstanceAnnotationValue(string annotatedPropertyNa
+                Debug.Assert(annotationNameFromReader == null, "annotationNameFromReader == null");
+
+                // Handle 'odata.XXXXX' annotations
+                if (isInstanceAnnotation && ODataJsonLightReaderUtils.IsODataAnnotationName(propertyNameFromReader))
+                {
+                    return Tuple.Create(PropertyParsingResult.ODataInstanceAnnotation, parsedPropertyName);
+                }
+
+                // Handle custom annotations
+                return Tuple.Create(PropertyParsingResult.CustomInstanceAnnotation, parsedPropertyName);
+            }
+
+            this.AssertJsonCondition(JsonNodeType.EndObject);
+            if (parsedPropertyName != null)
+            {
+                if (ODataJsonLightReaderUtils.IsAnnotationProperty(parsedPropertyName))
+                {
+                    throw new ODataException(
+                        Strings.ODataJsonLightDeserializer_AnnotationTargetingInstanceAnnotationWithoutValue(
+                            lastPropertyAnnotationNameFound,
+                            parsedPropertyName));
+                }
+
+                return Tuple.Create(PropertyParsingResult.PropertyWithoutValue, parsedPropertyName);
+            }
+
+            return Tuple.Create(PropertyParsingResult.EndOfObject, parsedPropertyName);
+        }
+
+        /// <summary>
+        /// Asynchronously process the current property annotation.
+        /// </summary>
+        /// <param name="annotatedPropertyName">The name being annotated. Can be a property or an instance annotation.</param>
+        /// <param name="annotationName">The annotation targeting the <paramref name="annotatedPropertyName"/>.</param>
+        /// <param name="propertyAndAnnotationCollector">The duplicate property names checker.</param>
+        /// <param name="readPropertyAnnotationValueDelegate">Delegate to read the property annotation value.</param>
+		/// <returns>A task that represents the asynchronous read operation.</returns>
+        private Task ProcessPropertyAnnotationAsync(
+            string annotatedPropertyName,
+            string annotationName,
+            PropertyAndAnnotationCollector propertyAndAnnotationCollector,
+            Func<string, Task<object>> readPropertyAnnotationValueDelegate)
+        {
+            // We don't currently support annotation targeting an instance annotation except for the @odata.type property annotation.
+            if (ODataJsonLightReaderUtils.IsAnnotationProperty(annotatedPropertyName)
+                && string.CompareOrdinal(annotationName, ODataAnnotationNames.ODataType) != 0)
+            {
+                return TaskUtils.GetFaultedTask<ODataException>(
+                    new ODataException(Strings.ODataJsonLightDeserializer_OnlyODataTypeAnnotationCanTargetInstanceAnnotation(
+                        annotationName,
+                        annotatedPropertyName,
+                        ODataAnnotationNames.ODataType)));

```

Comment:
Maybe you've explained this before, but why don't we use built-ins like `Task.FromException`?

Reply:
Gentle reminder just in case you missed this question.

Reply:
@habbes Because `Task.FromException` is not available in **net45** and **netstandard1.1**